### ISSUE TITLE
Should Syntax does not handle should.throws with a regex

### ISSUE THREAD

**Issue by yvele** [skovhus/jest-codemods#49]
This line:

```js
should.throws(foo, /^Error: Description/);
```

is transformed into:

```js
expect(foo).throws(/^Error: Description/);
```

but should be

```js
expect(foo).toThrowError(/^Description/);
```

PS: As a dirty workaround I just replace `).throws(/^Error: ` with `).toThrowError(/^`

**Comment by skovhus** [comment]
I get that `should.throws(foo` should be converted to `expect(foo).toThrowError(`, but I don't see why "Error:" should be removed.

I don't see this behaviour documented in chai or in their source code... Do you have a link?

http://chaijs.com/api/assert/ is the closets I get. :)

**Comment by skovhus** [comment]
And is `should.throws` supported with the `s`? In the docs it seems only `should.throw` is.

**Comment by yvele** [comment]
@skovhus Yep I it's with a `s` in the documentation https://shouldjs.github.io/#should-throws

**Comment by yvele** [comment]
Hum about the `Error: ` part I don't know.. that's really strange..

[should implementation](https://github.com/shouldjs/should.js/blob/5091f9c091a5df30ebc72c7c20af80b37e229bae/lib/ext/error.js#L65) looks the same as [jest implementation](https://github.com/facebook/jest/blob/5f72d73929980cedfaec58b7445210f87f233305/packages/jest-matchers/src/toThrowMatchers.js#L106)... hum.

**Comment by skovhus** [comment]
Should.js is actually not chai should. They have different assertions... But I'm happy that it almost worked for you. : )

The challenge is that a lot of people are not importing chai in their test files, but using mocha to set it up. So we cannot really know if the user actually uses chai should (or a similar library as should.js). 

I think I will insert a warning if should.js is imported/required. And then note that it might not be 100% supported yet. 

https://github.com/shouldjs/should.js/blob/master/Readme.md

**Comment by skovhus** [comment]
Let me know if I can improve the CLI or Readme to make it clear that chai should === should.js

**Comment by skovhus** [comment]
(Official) Should.js support released as 0.10.0. Let me know if it works for you. :)
